core/types/block.go

```diff
@@ -108,15 +108,16 @@ type Header struct {
 	// BaseFee was added by EIP-1559 and is ignored in legacy headers.
 	BaseFee *big.Int `json:"baseFeePerGas" rlp:"optional"`
 
-	// WithdrawalsHash was added by EIP-4895 and is ignored in legacy headers.
-	WithdrawalsHash *common.Hash `json:"withdrawalsRoot" rlp:"optional"`
-
 	// BlobGasUsed was added by EIP-4844 and is ignored in legacy headers.
 	BlobGasUsed *uint64 `json:"blobGasUsed" rlp:"optional"`
 
 	// ExcessBlobGas was added by EIP-4844 and is ignored in legacy headers.
 	ExcessBlobGas *uint64 `json:"excessBlobGas" rlp:"optional"`
 
+	// Leave fields not used by BSC last to avoid being serialized by RLP
+	// WithdrawalsHash was added by EIP-4895 and is ignored in legacy headers.
```

Comment:
We should mention in the comment here that WithdrawalsHash is always `nil` for bsc/parlia to avoid confusion in future.